src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -0,0 +1,320 @@
+            {
+                if (runtimeFeatureType != null && HasPreviewAttribute(runtimeFeatureType, requiresPreviewFeaturesSymbols))
+                {
+                    requiresPreviewFeaturesSymbolsToUsageType.GetOrAdd(propertyOrMethodSymbol, PreviewFeatureUsageType.StaticAbstract);
+                    return true;
+                }
+            }
+
+            if (propertyOrMethodSymbol.IsImplementationOfAnyImplicitInterfaceMember(out ISymbol baseInterfaceMember))
+            {
+                if (HasPreviewAttribute(baseInterfaceMember, requiresPreviewFeaturesSymbols) || ContainingTypeIsPreview(baseInterfaceMember, requiresPreviewFeaturesSymbols))
+                {
+                    return true;
+                }
+            }
+
+            if (propertyOrMethodSymbol.IsOverride)
+            {
+                ISymbol? overridden = propertyOrMethodSymbol.GetOverriddenMember();
+                if (overridden != null && HasPreviewAttribute(overridden, requiresPreviewFeaturesSymbols))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static void AnalyzeSymbol(SymbolAnalysisContext context, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, ConcurrentDictionary<ISymbol, PreviewFeatureUsageType> requiresPreviewFeaturesSymbolsToUsageType, INamedTypeSymbol? runtimeFeatureType)
+        {
+            ISymbol symbol = context.Symbol;
+            if (symbol == null)
+            {
+                return;
+            }
+
+            // Don't report diagnostics on types that are marked as Preview
+            if (HasPreviewAttribute(symbol, requiresPreviewFeaturesSymbols))
+            {
+                return;
+            }
+
+            bool reportDiagnostic = false;
+
+            // Look for attributes on base classes that this type may extend
+            if (symbol is ITypeSymbol typeSymbol)
+            {
+                reportDiagnostic = TypeSymbolHasAttribute(typeSymbol, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType);
+            }
+
+            if (!reportDiagnostic)
+            {
+                if (symbol is IMethodSymbol || symbol is IPropertySymbol)
+                {
+                    reportDiagnostic = PropertyOrMethodSymbolHasAttribute(symbol, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, runtimeFeatureType);
+                }
+            }
+
+            if (reportDiagnostic)
+            {
+                if (requiresPreviewFeaturesSymbolsToUsageType.TryGetValue(symbol, out PreviewFeatureUsageType usageType))
+                {
+                    if (usageType == PreviewFeatureUsageType.StaticAbstract)
+                    {
+                        context.ReportDiagnostic(symbol.CreateDiagnostic(StaticAbstractIsPreviewFeatureRule));
```

Comment:
Will that get reported for something like:

```csharp
public abstract class C
{
    public static abstract M();
}
```

I think it shouldn't, since the above is an error anyway (static abstract is only allowed in interfaces), and this diagnostic will make it feel like this is allowed in preview.

Reply:
It will yea. However, I am unable to test it at the moment because we don't consume the latest compiler yet. Without testing, I'm not sure how the parent chain will look here. I've added a TODO for now. Once we started consuming the latest compiler here, I can fix this.

Reply:
Can you file a tracking issue for this and assign it to yourself?